File: IdealWeatherAPI/Layers/Application/Repositories/Geocoding/USCensus/Validations/AddressValidator.cs
using IdealWeatherAPI.Layers.Domain.Geocoding;

namespace IdealWeatherAPI.Layers.Application.Repositories.Geocoding.USCensus.Validations
{
    public class AddressValidator
    {
        private readonly Address _address;

        public AddressValidator(Address address)
        {
            _address = address;
        }

        /// <summary>
        /// Validates the address for use with the US Census Geocoding Service's "address" search type. 
        /// </summary>
        /// <returns>True if the street and postal code OR the street, city, and state are present.</returns>
        public bool AddressIsValid()
        {
            if (!string.IsNullOrWhiteSpace(_address.StreetName) && !string.IsNullOrWhiteSpace(_address.Postalcode))
            {
                return true;
            }

            if (!string.IsNullOrWhiteSpace(_address.StreetName) && !string.IsNullOrWhiteSpace(_address.City) && !string.IsNullOrWhiteSpace(_address.State))
            {
                return true;
            }

            return false;
        }

    }
}

File: weather-station-2001/src/apis/weather_client/weather_services/LocalWeatherService/AddressValidator.ts
import type { TAddress } from "../../WeatherClient.types"

export class AddressValidator {
	constructor(private address: TAddress) {}

	validate() {
		if (!!this.address.streetName?.trim() && !!this.address.postalCode?.trim()) {
			return true
		}

		if (!!this.address.streetName?.trim() && !!this.address.city?.trim() && !!this.address.state?.trim()) {
			return true
		}

		return false
	}
}
